agent/structs/check_type.go

```diff
@@ -91,6 +93,17 @@ func (t *CheckType) UnmarshalJSON(data []byte) (err error) {
 	}{
 		Alias: (*Alias)(t),
 	}
+
+	// Preevaluate struct values to determine where to set defaults
+	if err = lib.UnmarshalJSON(data, aux); err != nil {
+		return err
+	}
```

Comment:
I guess we could do this unmarshal with a target that only has the `H2PING` fields, to make it slightly more explicit that we are only checking if that setting is enabled, but I think that's a minor thing, not blocking, could be a follow up.